File: dodam-system-domain/dodam-domain-rds/src/main/java/b1nd/dodam/domain/rds/member/repository/DormitoryManageMemberRepository.java
package b1nd.dodam.domain.rds.member.repository;

import b1nd.dodam.domain.rds.member.entity.DormitoryManageMember;
import b1nd.dodam.domain.rds.member.entity.Member;
import b1nd.dodam.domain.rds.member.exception.DormitoryManageMemberNotFoundExcpetion;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface DormitoryManageMemberRepository extends JpaRepository<DormitoryManageMember, Integer> {

    default DormitoryManageMember getByMember(Member member) {
        return findByMember(member)
                .orElseThrow(DormitoryManageMemberNotFoundExcpetion::new);
    }

    Optional<DormitoryManageMember> findByMember(Member member);

    boolean existsByMember(Member member);

    void deleteByMember(Member member);

}

File: services/service-member/src/main/kotlin/com/b1nd/dodamdodam/member/domain/dormitory/repository/DormitoryManageMemberRepository.kt
package com.b1nd.dodamdodam.member.domain.dormitory.repository

import com.b1nd.dodamdodam.member.domain.dormitory.entity.DormitoryManageMemberEntity
import com.b1nd.dodamdodam.member.domain.member.entity.MemberEntity
import org.springframework.data.jpa.repository.JpaRepository

interface DormitoryManageMemberRepository : JpaRepository<DormitoryManageMemberEntity, Long> {
    fun findByMember(member: MemberEntity): DormitoryManageMemberEntity?
    fun existsByMember(member: MemberEntity): Boolean
}
